discord_bots/views/configure_category.py

```diff
@@ -0,0 +1,319 @@
+import logging
+
+from discord import (
+    ButtonStyle,
+    Client,
+    Colour,
+    Embed,
+    Interaction,
+    SelectOption,
+    TextStyle,
+)
+from discord.ui import button, Button, Modal, Select, TextInput
+
+from discord_bots.models import Category
+from discord_bots.views.base import BaseView
+from discord_bots.views.confirmation import ConfirmationView
+
+_log = logging.getLogger(__name__)
+
+
+class CategoryConfigureView(BaseView):
+    def __init__(self, interaction: Interaction, category: Category):
+        super().__init__(timeout=300)
+        self.value: bool = False
+        self.category: Category = category
+        self.interaction: Interaction = interaction
+        self.embed: Embed
+        self.add_item(CategoryRatedSelect(self))
+
+    @button(label="Set Name", style=ButtonStyle.primary, row=0)
+    async def setname(self, interaction: Interaction, button: Button):
+        modal = CategoryNameModal(self)
+        await interaction.response.send_modal(modal)
+        return True
+
+    @button(label="Leaderboard Settings", style=ButtonStyle.primary, row=0)
+    async def leaderboard(self, interaction: Interaction, button: Button):
+        modal = CategoryLeaderboardModal(self)
+        await interaction.response.send_modal(modal)
+        return True
+
+    @button(label="Decay Settings", style=ButtonStyle.primary, row=0)
+    async def decay(self, interaction: Interaction, button: Button):
+        modal = CategoryDecayModal(self)
+        await interaction.response.send_modal(modal)
+        return True
+
+    @button(label="Save", style=ButtonStyle.success, row=4)
+    async def save(self, interaction: Interaction, button: Button):
+        await interaction.response.defer()
+        await self.disable_children(interaction)
+
+        confirmation_buttons = ConfirmationView(interaction.user.id)
+        confirmation_buttons.message = await interaction.followup.send(
+            embed=Embed(
+                description=f"⚠️ Are you sure you want to save configurations for category **{self.category.name}**?⚠️",
+                colour=Colour.yellow(),
+            ),
+            view=confirmation_buttons,
+            ephemeral=True,
+        )
+        await confirmation_buttons.wait()
+        if not confirmation_buttons.value:
+            await self.enable_children(interaction)
+            return False
+        else:
+            self.value = True
+            self.stop()
```

Comment:
Should saving completely stop the view? What if I want to set the name, save, and then mess around with the sigma decay settings? If I'm understanding this correctly, the view will be completely stopped with buttons disabled after saving the settings.